llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -3917,3 +3903,52 @@ def atomic_thread_fence_seq_cst_cta :
 def atomic_thread_fence_acq_rel_cta :
   NVPTXInst<(outs), (ins), "fence.acq_rel.cta;", []>,
   Requires<[hasPTX<60>, hasSM<70>]>;
+
+def fpimm_any_zero : FPImmLeaf<fAny, [{
+  return Imm.isExactlyValue(+0.0) | Imm.isExactlyValue(-0.0);
```

Comment:
This might give a warning about bitwise orbeing used instead of logical or. I'd suggest using `||`.

Alternatively, perhaps `Imm.isExactlyValue(+0.0) | Imm.isExactlyValue(-0.0)` is just `Imm.isZero()`?

Reply:
Have updated